fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -1951,31 +1951,70 @@ class DDTeamCollectionImpl {
 		return Void();
 	}
 
+	ACTOR static Future<double> loadBytesBalanceRatio(DDTeamCollection* self) {
+		wait(delayJittered(SERVER_KNOBS->CHECK_LOAD_BYTES_BALANCE_DELAY)); // at most 3 min simulation
+		double minLoadBytes = std::numeric_limits<double>::max();
+		double avgLoadBytes = 0;
+		int count = 0;
+		for (auto& [id, s] : self->server_info) {
+			// If a healthy SS don't have storage metrics, skip this round
+			if (self->server_status.get(s->getId()).isUnhealthy()) {
+				continue;
+			} else if (!s->metricsPresent()) {
+				return 0;
+			}
+
+			double load = s->loadBytes();
+			avgLoadBytes += load;
+			++count;
+			minLoadBytes = std::min(minLoadBytes, load);
+		}
+		avgLoadBytes /= count;
+		return minLoadBytes / avgLoadBytes;
+	}
+
+	ACTOR static Future<Void> perpetualStorageWiggleRest(DDTeamCollection* self) {
+		state Future<Void> maxDelay = delay(SERVER_KNOBS->PERPETUAL_WIGGLE_DELAY);
+		state bool takeRest = true;
+		while (takeRest) {
```

Comment:
This actor can be simplified/changed to something like this, to satisfy the configured delay and then the load balance requirement and stop immediately after observing that the load balance and server/machine counts are good.

```
wait(delay(SERVER_KNOBS->PERPETUAL_WIGGLE_DELAY));

loop {
  wait(delay(SERVER_KNOBS->CHECK_LOAD_BYTES_BALANCE_DELAY));
  balance = self->loadBalanceBytesRatio();

  ... your takeRest logic here, break when rest is no longer needed
}
```